system/diagnostic_graph_aggregator/src/common/graph/loader.cpp

```diff
@@ -59,7 +59,7 @@ UnitLink * UnitLoader::child() const
 std::vector<UnitLink *> UnitLoader::children() const
 {
   std::vector<UnitLink *> result;
-  for (const auto & config : config_->list) {
+  for (auto * const config : config_->list) {
```

Comment:
Is it OK to use `auto * const` instead of `const auto * const`?

Reply:
The value itself cannot be const because it must be cast to a modifiable reference.

```
/home/emb4/autoware_cppcheck/autoware/src/universe/autoware.universe/system/diagnostic_graph_aggregator/src/common/graph/loader.cpp:63:45: error: binding reference of type ‘diagnostic_graph_aggregator::LinkConfig* const&’ to ‘const diagnostic_graph_aggregator::LinkConfig* const’ discards qualifiers
   63 |     result.push_back(links_.config_links.at(config));
      |                                             ^~~~~~
```

Reply:
Yes, you can use it.

`const auto * const` makes both pointer and value const.
In contrast, `auto * const`  consts only the pointer and allows the value to be changed.

In this case `auto * const` is correct because we need to pass config as a modifiable reference as follows

```
universe/autoware.universe/system/diagnostic_graph_aggregator/src/common/graph/loader.cpp:63:45: error: binding reference of type ‘diagnostic_graph_aggregator::LinkConfig* const&’ to ‘const diagnostic_graph_aggregator::LinkConfig* const’ discards qualifiers
   63 |     result.push_back(links_.config_links.at(config));
      |                                             ^~~~~~
```

Reply:
Thank you.